src/main/java/org/altbeacon/beacon/service/GattBeaconTracker.java

```diff
@@ -69,13 +62,12 @@ private Beacon trackGattBeacon(Beacon beacon) {
     private void updateTrackingHashes(Beacon trackedBeacon, HashMap<Integer,Beacon> matchingTrackedBeacons) {
         if (matchingTrackedBeacons == null) {
             matchingTrackedBeacons = new HashMap<Integer,Beacon>();
-            mBeaconsByServiceUuidAndMac.put(serviceUuidAndMac(trackedBeacon), matchingTrackedBeacons);
         }
-        mBeacons.put(trackedBeacon.hashCode(), trackedBeacon);
         matchingTrackedBeacons.put(trackedBeacon.hashCode(), trackedBeacon);
+        mBeaconsByKey.put(getBeaconKey(trackedBeacon), matchingTrackedBeacons);
     }
 
-    private String serviceUuidAndMac(Beacon beacon) {
-        return beacon.getBluetoothAddress()+beacon.getServiceUuid();
+    private String getBeaconKey(Beacon beacon) {
+        return beacon.getBluetoothAddress() + beacon.getServiceUuid();
```

Comment:
I thought about removing the serviceUUID from the "beaconKey" which would make the beacons unique only by their MAC address, but I don't know if this would be ok. What do you think?